### DISCUSSION TITLE
April Fool's 2022 Changes???

### DISCUSSION THREAD

**Discussion by ptarsus07** [apprenticeharper/DeDRM_tools#2045] [Q&A]
As recently as 03/17/22, I was able to successfully remove DRM from new Kindle purchases; however, today (04/01/2022), I started getting errors.  Does anyone know if something has changed in the last two weeks?

Hardware =
-- MacBook Air (M1, 2020)
-- macOS Monterey (v12.2.1)
Software =
-- K4M v1.31 (to download the newly-purchased AZW file(s))
-- Calibre v4.23.0 (to edit metadata & run conversion)
-- Apprentice Alf's DeDRM tools (v6.8.0) (to remove DRM)

I am attempting to convert from AZW to EPUB, and I'm getting an error that reads:

Cannot convert [title]
This book is protected by **DRM**.  To learn more about DRM and why you cannot read or convert this book in calibre, [click here](https://manual.calibre-ebook.com/drm.html).

As noted, I've been using this configuration successfully for a long time and as recently as two weeks ago.

Any help/ideas will be greatly appreciated...

**Comment by ptarsus07** [comment]
Reading more at Apprentice Alf's blog, I realize I should post a log, so here it is:

calibre Debug log
calibre 4.23  embedded-python: True is64bit: True
Darwin-21.3.0-x86_64-i386-64bit Darwin ('64bit', '')
('Darwin', '21.3.0', 'Darwin Kernel Version 21.3.0: Wed Jan  5 21:37:58 PST 2022; root:xnu-8019.80.24~20/RELEASE_ARM64_T8101')
Python 2.7.16
OSX: ('10.16', ('', '', ''), 'x86_64')
Interface language: None
Successfully initialized third party plugins: DeDRM (6, 8, 0)
calibre 4.23  embedded-python: True is64bit: True
Darwin-21.3.0-x86_64-i386-64bit Darwin ('64bit', '')
('Darwin', '21.3.0', 'Darwin Kernel Version 21.3.0: Wed Jan  5 21:37:58 PST 2022; root:xnu-8019.80.24~20/RELEASE_ARM64_T8101')
Python 2.7.16
OSX: ('10.16', ('', '', ''), 'x86_64')
Interface language: None
Successfully initialized third party plugins: DeDRM (6, 8, 0)
devicePixelRatio: 2.0
logicalDpi: 72.0 x 72.0
physicalDpi: 127.500001915 x 127.500001915
Using calibre Qt style: True
[0.00] Starting up...
[0.01] Showing splash screen...
[0.17] splash screen shown
[0.17] Initializing db...
[0.19] db initialized
[0.19] Constructing main UI...
[0.96] main UI initialized...
[0.96] Hiding splash screen
[0.97] splash screen hidden
[0.97] Started up in 0.97 seconds with 520 books
DeDRM v6.8.0: Trying to decrypt B00I5QX1A2_EBOK.azw
Using Library AlfCrypto Python
MobiDeDrm v0.42.
Copyright © 2008-2017 The Dark Reverser, Apprentice Harper et al.
Decrypting Kindle Format 8 ebook: The Confession
Got DSN key from database default_key
Got DSN key from database default_key_2
Found 8 keys to try after 0.0 seconds
Crypto Type is: 2
DeDRM v6.8.0: Failed to decrypt with error: No key found in 8 keys tried.
DeDRM v6.8.0: Looking for new default Kindle Key after 0.0 seconds
Found k4Mac kinf2018 file: /Users/[user]/Library/Application Support/Kindle/storage/.kinf2018
Found k4Mac kinf2011 file: /Users/[user]/Library/Application Support/Kindle/storage/.kinf2011
Couldn't decrypt file.
Couldn't decrypt file.
DeDRM v6.8.0: Ultimately failed to decrypt after 0.2 seconds. Read the FAQs at Harper's repository: https://github.com/apprenticeharper/DeDRM_tools/blob/master/FAQs.md
Running file type plugin DeDRM failed with traceback:
Traceback (most recent call last):
  File "site-packages/calibre/customize/ui.py", line 172, in _run_filetype_plugins
  File "calibre_plugins.dedrm.__init__", line 625, in run
  File "calibre_plugins.dedrm.__init__", line 575, in KindleMobiDecrypt
DeDRMError: DeDRM v6.8.0: Ultimately failed to decrypt after 0.2 seconds. Read the FAQs at Harper's repository: https://github.com/apprenticeharper/DeDRM_tools/blob/master/FAQs.md
Added The Confession to db in: 0.0
Added 1 books in 0.6 seconds

**Reply by ZolaLa9** [reply]
Maybe try DeDRM v [6.8.1](https://github.com/apprenticeharper/DeDRM_tools/releases/tag/v6.8.1) or [10.0.2](https://github.com/noDRM/DeDRM_tools/releases)?

**Reply by ptarsus07** [reply]
Thank you, ZolaLa9, but...

DeDRM v6.8.1 appears to be for users on macOS Big Sur (v11); I'm running macOS Monterey (v12).  Apprentice Harper's 11/28/21 release note says, "you only need this update if you use Kindle for Mac under Mac OS X Big Sur."  Not sure I want to install it if it doesn't apply, especially since my current installation was working two weeks ago (some time after 6.8.1 was released).

DeDRM v10.0.2 appears to be correcting an typo in the (previous?) kindlekey.py file that "breaks key retrieval, causing decryption on new installations of Kindle for Mac using DeDRM 7.2.1 to fail."  Since I'm not using DeDRM 7.2.1, I'm unsure this would address my issue, either (and for the same reasons as above).

I'm not a coder, so I'm not sure I'm reading the log correctly, but it appears the error is happening here:

Got DSN key from database default_key
Got DSN key from database default_key_2
Found 8 keys to try after 0.0 seconds
Crypto Type is: 2
DeDRM v6.8.0: Failed to decrypt with error: No key found in 8 keys tried.
DeDRM v6.8.0: Looking for new default Kindle Key after 0.0 seconds
Found k4Mac kinf2018 file: /Users/[user]/Library/Application Support/Kindle/storage/.kinf2018
Found k4Mac kinf2011 file: /Users/[user]/Library/Application Support/Kindle/storage/.kinf2011

Possibly related to "Crypto Type is: 2" ???

**Reply by ElleKayEm** [reply]
I'd also suggest updating to the latest calibre.  In which case, you will need DeDRM 10.0.2.
